### PR TITLE
♻️  `webserver`: fixes mypy issues in `resource-manager` plugin

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ♻️ fixes mypy issues in ``webserver`` plugin ``resource_manager``
- ♻️ fixes all mypy issues in `webserver` except for
```cmd
make mypy | grep -v -E "exporter|projects|rest_"
```
- @matusdrobuliak66 can you please take care of `projects` after this PR is in?
- @GitHK  can you please take care of `exporter` after this PR is in?
- @pcrespov will take care of `rest` as soon as https://github.com/ITISFoundation/osparc-simcore/pull/4316 merges.

## Related issue/s

- part of https://github.com/ITISFoundation/osparc-simcore/issues/4071

## How to test

```cmd
cd services/web/server
make install-dev
make mypy | grep -v -E "exporter|projects|rest_"
```

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
nice :+1: yes i will take care of the projects plugin

File: services/web/server/src/simcore_service_webserver/application_settings.py [hunk 1]
```diff
@@ -375,13 +376,16 @@ def to_client_statics(self) -> dict[str, Any]:
 
 
 def setup_settings(app: web.Application) -> ApplicationSettings:
-    app[APP_SETTINGS_KEY] = settings = ApplicationSettings.create_from_envs()
-    log.info(
+    settings: ApplicationSettings = ApplicationSettings.create_from_envs()
+    app[APP_SETTINGS_KEY] = settings
+    log.debug(
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/application_settings.py:381]: maybe also apply the `_logger` rename here as well
```suggestion
    _logger.debug(
```

File: services/web/server/src/simcore_service_webserver/long_running_tasks.py [hunk 2]
```diff
@@ -17,8 +17,8 @@
 
 
 class _RequestContext(BaseModel):
-    user_id: UserID = Field(..., alias=RQT_USERID_KEY)
-    product_name: str = Field(..., alias=RQ_PRODUCT_KEY)
+    user_id: UserID = Field(..., alias=RQT_USERID_KEY)  # type:ignore
+    product_name: str = Field(..., alias=RQ_PRODUCT_KEY)  # type:ignore
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/long_running_tasks.py:21]: Q: why are these ignored?

File: services/web/server/src/simcore_service_webserver/resource_manager/registry.py [hunk 3]
```diff
@@ -71,8 +71,8 @@ def _decode_hash_key(cls, hash_key: str) -> RegistryKeyPrefixDict:
             if hash_key.endswith(f":{RESOURCE_SUFFIX}")
             else hash_key[: -len(f":{ALIVE_SUFFIX}")]
         )
-        key = dict(x.split("=") for x in tmp_key.split(":"))
-        return RegistryKeyPrefixDict(**key)
+        key = dict(part.split("=") for part in tmp_key.split(":"))
+        return RegistryKeyPrefixDict(**key)  # type: ignore
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/resource_manager/registry.py:75]: Q: why is this one not liked?

**Review by GitHK — APPROVED** [review]

**Comment by GitHK** [comment]
@pcrespov I will try to put some work into the exporter this sprint.

**Review by sanderegg — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/long_running_tasks.py (hunk 2 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/long_running_tasks.py:21]:
```cmd
src/simcore_service_webserver/long_running_tasks.py:20:5: error: Required dynamic aliases disallowed  [pydantic-alias]
src/simcore_service_webserver/long_running_tasks.py:21:5: error: Required dynamic aliases disallowed  [pydantic-alias]
```

But in reality they are not dynamic, they are `Final` constants

File: services/web/server/src/simcore_service_webserver/resource_manager/registry.py (hunk 3 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/resource_manager/registry.py:75]: RegistryKeyPrefixDict is not total but at the same time he does not like **key

**Pull request merged**